ddl/column.go

```diff
@@ -585,17 +588,20 @@ func (w *worker) onModifyColumn(t *meta.Meta, job *model.Job) (ver int64, _ erro
 	oldColName := &model.CIStr{}
 	pos := &ast.ColumnPosition{}
 	var modifyColumnTp byte
-	err := job.DecodeArgs(newCol, oldColName, pos, &modifyColumnTp)
+	var updatedAutoRandomBits uint64
+	err := job.DecodeArgs(newCol, oldColName, pos, &modifyColumnTp, &updatedAutoRandomBits)
```

Comment:
If this DDL operation needs to be handled during the rolling upgrade, there may be compatibility issues.

Reply:
If modify `auto_random` bits is really necessary during updates, we need to consider two cases:
- The DDL owner is an old-version TiDB: `auto_random` shard bits remain unchanged(but DDL successfully executed).
- The DDL owner is a new-version TiDB: the job arguments mismatch, throwing an error.

I think none of them can lead to data inconsistency.

Reply:
> The DDL owner is an old-version TiDB: auto_random shard bits remain unchanged(but DDL successfully executed).

It will not cause data inconsistency, but it will confuse the user, the execution is successful, but the function is not successful.

Reply:
How can I add job arguments to DDL jobs without affecting the rolling update?

Reply:
At the moment, I don’t seem to have any good methods other than the newly added types. Or you ask if the DBA can accept this situation

Reply:
LGTM

Reply:
Ok, we can write this in the documents.